src/types/redis_zset.cc

```diff
@@ -71,19 +72,35 @@ rocksdb::Status ZSet::Add(const Slice &user_key, uint8_t flags, std::vector<Memb
     }
     added_member_keys.insert(member_key);
 
+    bool lt = (flags & kZSetLT) != 0;
+    bool gt = (flags & kZSetGT) != 0;
+
     if (metadata.size > 0) {
       std::string old_score_bytes;
       s = db_->Get(rocksdb::ReadOptions(), member_key, &old_score_bytes);
       if (!s.ok() && !s.IsNotFound()) return s;
       if (s.ok()) {
+        if (!s.IsNotFound() && (flags & kZSetNX)) {
+          continue;
+        }
         double old_score = DecodeDouble(old_score_bytes.data());
-        if (flags == kZSetIncr) {
+        if (flags & kZSetIncr) {
+          if (lt && (*mscores)[i].score >= 0) {
+            continue;
+          } else if (gt && (*mscores)[i].score <= 0) {
+            continue;
+          }
```

Comment:
```suggestion
          if (lt && (*mscores)[i].score >= 0) {
            continue;
          } else if (gt && (*mscores)[i].score <= 0) {
            continue;
          }
```
Can merge into one line with `||`

Reply:
merged